packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -38,6 +38,26 @@ class BuildWebCommand extends BuildSubCommand {
       help: 'Disable dynamic generation of code in the generated output. '
         'This is necessary to satisfy CSP restrictions (see http://www.w3.org/TR/CSP/).'
     );
+    argParser.addOption('pwa-strategy',
+      defaultsTo: kOfflineFirst,
+      help:
+        'The caching strategy to be used by the PWA service worker.\n'
+        'offline-first will attempt to cache the app shell eagerly and '
+        'then lazily cache all subsequent assets as they are loaded. When '
+        'making a network request for an asset, the offline cache will be '
+        'preferred.\n'
+        'online-first will always attempt to make a network request for an '
+        'asset, including the app shell. If this fails for any reason, the '
+        'app will fallback on a cached version if available. This cache will '
+        'be lazily populated as resources are requested.\n'
+        'online-only will never attempt to perform offline caching of resources. '
+        'if there is no network connectivity, the application will fail to load.',
```

Comment:
Would be good to spend a sentence for each option on explaining the use-case. For example: "online-only is useful if you don't want any local caching but still support adding/installing the app on the home screen as a PWA."